packages/data-stores/src/plans/resolvers.ts

```diff
@@ -41,90 +40,96 @@ function getMonthlyPrice( plan: APIPlan ) {
-	Business: PLAN_BUSINESS,
-	eCommerce: PLAN_ECOMMERCE,
-};
-
-export function* getPlansDetails( locale = 'en' ) {
-	try {
-		const { body: rawPlansDetails } = yield fetchAndParse(
-			`https://public-api.wordpress.com/wpcom/v2/plans/details?locale=${ encodeURIComponent(
-				locale
-			) }`,
-			{
-				mode: 'cors',
-				credentials: 'omit',
-			}
-		);
-
-		const plans: Record< string, Plan > = {};
-		const features: Record< string, PlanFeature > = {};
-
-		rawPlansDetails.features.forEach( ( feature: Record< string, string > ) => {
-			features[ feature.id ] = {
-				id: feature.id,
-				name: feature.name,
-				description: feature.description,
-				type: feature.type ?? 'checkbox',
-			};
-		} );
-
-		rawPlansDetails.plans.forEach( ( rawPlan: APIPlanDetail ) => {
-			const plan: Plan = {
-				title: rawPlan.short_name,
-				description: rawPlan.tagline,
-				productId: rawPlan.products[ 0 ].plan_id,
-				storeSlug: mapShortNameToProductSlug[ rawPlan.nonlocalized_short_name ],
-				features: rawPlan.highlighted_features,
-				pathSlug: rawPlan.nonlocalized_short_name?.toLowerCase(),
-				featuresSlugs: rawPlan.features.reduce(
-					( acc: Record< string, boolean >, cur: string ) => ( { ...acc, [ cur ]: true } ),
-					{}
-				),
-				storage: rawPlan.storage,
-			};
-
-			if ( 'Free' === rawPlan.nonlocalized_short_name ) {
-				plan.isFree = true;
-			}
-
-			if ( 'Premium' === rawPlan.nonlocalized_short_name ) {
-				plan.isPopular = true;
-			}
+	const { body: plansFeatures } = ( yield fetchAndParse(
+		`https://public-api.wordpress.com/wpcom/v2/plans/details?locale=${ encodeURIComponent(
+			locale
+		) }`,
+		{
+			mode: 'cors',
+			credentials: 'omit',
+		}
+	) ) as { body: DetailsAPIResponse };
+
+	const periodAgnosticPlans: Plan[] = plansFeatures.plans.map( ( plan ) => {
+		return {
+			description: plan.tagline,
+			features: plan.highlighted_features,
+			storage: plan.storage,
+			title: plan.short_name,
+			featuresSlugs: plan.features.reduce( ( slugs, slug ) => {
+				slugs[ slug ] = true;
+				return slugs;
+			}, {} as Record< string, boolean > ),
+			isFree: plan.nonlocalized_short_name === TIMELESS_PLAN_FREE,
+			isPopular: plan.nonlocalized_short_name === TIMELESS_PLAN_PREMIUM,
+			periodAgnosticSlug: plan.nonlocalized_short_name,
+			productIds: plan.products.map( ( { plan_id } ) => plan_id ),
+		};
+	} );
 
-			plans[ mapShortNameToProductSlug[ rawPlan.nonlocalized_short_name ] ] = plan;
+	const planProducts: PlanProduct[] = plansProductSlugs.reduce( ( plans, slug ) => {
+		const planProduct = pricedPlans.find(
+			( pricedPlan ) => pricedPlan.product_slug === slug
+		) as PricedAPIPlan;
+
+		if ( ! planProduct ) {
+			return plans;
+		}
+
+		const periodAgnosticPlan = periodAgnosticPlans.find(
+			( plan ) => plan.productIds.indexOf( planProduct.product_id ) > -1
+		) as Plan;
+
+		plans.push( {
+			productId: planProduct.product_id,
+			billingPeriod: planProduct.bill_period === 31 ? 'MONTHLY' : 'ANNUALLY',
```

Comment:
Don't we need to store this somewhere globally? From my experience, hard-coded values and their purposes can be forgotten very easily.

Reply:
It shouldn't be an issue because it's [typed](https://github.com/Automattic/wp-calypso/blob/cff5fabf1fed47ee7741ef6729a4c191e81b0664/packages/data-stores/src/plans/types.ts#L27).

Reply:
Exactly, and [`planProduct.bill_period`](https://github.com/Automattic/wp-calypso/pull/48790/files#diff-d822933ae1c209e261cde32fc0c43cf346c6271be748990c5c5edd5c289f48bcR86) as well but it's unclear what the hard-coded number actually represents. I assume 31 represents the length of the month and there probably is a similar (i.e. 365) value for the length of a year but that's not clear. Besides that, the hard-coded value is reused on [line 98](https://github.com/Automattic/wp-calypso/pull/48790/files#diff-658e8069f3b8ab48bcd6560541fdf2d3021ee0d4e2d9d9a3a1be8ce88bd5bfcbR98), rendering it an excellent candidate to be stored in a variable.